Fix GitHub issues #32882-3: Surprising flow with Not Boolean? (#37773)
dotnet/docs: docs/visual-basic/language-reference/operators/not-operator.md

> [!NOTE]
> Since the logical and bitwise operators have a lower precedence than other arithmetic and relational operators, any bitwise operations should be enclosed in parentheses to ensure accurate execution.  
  

> Note that `if Not someStr?.Contains("some string")` or any other value that evaluates as `Boolean`? and has the value of `nothing` or `HasValue=false` should actually run an `else` block in `if` to prevent any confusion. The evaluation follows the SQL evaluation where null/nothing doesn't equal anything, not even another null/nothing.

Reviewer: Just a few small nits:

```suggestion
Note that `if Not someStr?.Contains("some string")` or any other value that evaluates as `Boolean?` and has the value of `nothing` or `HasValue=false` the `else` block is run.  The evaluation follows the SQL evaluation where null/nothing doesn't equal anything, not even another null/nothing.
```
